### PR TITLE
Remove '\r' from processed txt spiffs files

### LINKED ISSUE #386: Text spiffs processor should convert CRLF to LF
To prevent different behavior between Linux and Windows builds, the text spiffs processor should standardize line endings to UNIX-style when it's generating the spiffs-image.

Solving this problem with git will create more problems than it will solve, since certain git commands treat line-endings and line-ending-related configs differently, or only apply the clean or skew filter instead of both of them.

### PR DESCRIPTION
**Pull request by Brycey92**
Closes #386

### CODE REVIEW

File: spiffs_file_preprocessor/txt_processor.c [hunk 1]
```diff
@@ -7,6 +7,20 @@
 #include "heatshrink_encoder.h"
 #include "fileUtils.h"
 
+long remove_chars(char* str, long len, char c) {
```
Comment (AEFeinstein) [inline spiffs_file_preprocessor/txt_processor.c:10]: It is unclear whether `len` is the `strlen` (excluding terminator) or `sizeof` (including null terminator). Looking at how the function is used, I think it's the `strlen` size, but the loop also checks the null terminator. Versatile? You should make a doxygen comment with what args should be and what behavior is.

File: spiffs_file_preprocessor/txt_processor.c [hunk 2]
```diff
@@ -7,6 +7,20 @@
...
+    long newLen = 1;
+    for(long i = 0; i < len && *strReadPtr; i++)
+    {
+        *strWritePtr = *strReadPtr++;
+        strWritePtr += (*strWritePtr != c);
+        newLen += (*strWritePtr != c);
+    }
+    *strWritePtr = '\0';
+
+    return newLen;
```
Comment (AEFeinstein) [inline spiffs_file_preprocessor/txt_processor.c:21]: How did you test this? I spun up a trivial test program, and the length seems incorrect.
```C
#include <stdio.h>
#include <string.h>

long remove_chars(char* str, long len, char c) {
    char *strReadPtr = str, *strWritePtr = str;
    long newLen = 1;
    for(long i = 0; i < len && *strReadPtr; i++)
    {
        *strWritePtr = *strReadPtr++;
        strWritePtr += (*strWritePtr != c);
        newLen += (*strWritePtr != c);
    }
    *strWritePtr = '\0';

    return newLen;
}

int main()
{
    char rmv = 't';
    char tst[] = "test string";
    printf("remove '%c' from \"%s\"\n", rmv, tst);
    long newlen = remove_chars(tst, strlen(tst), rmv);
    printf("returned len: %d, strlen(): %d, str: \"%s\"\n", newlen, strlen(tst), tst);
    return 0;
}
```
prints
```
remove 't' from "test string"
returned len: 7, strlen(): 8, str: "es sring"
```

**Review by AEFeinstein — CHANGES_REQUESTED** [review]

File: spiffs_file_preprocessor/txt_processor.c (hunk 1 above)
Comment (Brycey92) [reply spiffs_file_preprocessor/txt_processor.c:10]: Done

File: spiffs_file_preprocessor/txt_processor.c (hunk 2 above)
Comment (Brycey92) [reply spiffs_file_preprocessor/txt_processor.c:29]: It should be 9, so including the null terminator. There was a bug where a couple of lines were swapped, which caused chars to not be counted sometimes. Fixed and pushed.

File: spiffs_file_preprocessor/txt_processor.c [hunk 3]
```diff
@@ -50,4 +58,10 @@ void process_txt(const char *infile, const char *outdir)
     FILE* outFile = fopen(outFilePath, "wb");
     fwrite(txtInStr, newSz, 1, outFile);
     fclose(outFile);
+
+    char rmv = 't';
+    char tst[] = "test string";
```
Comment (AEFeinstein) [inline spiffs_file_preprocessor/txt_processor.c:63]: I don't think you meant to leave this in, but I like that you used it to test

**Review by AEFeinstein — CHANGES_REQUESTED** [review]

File: spiffs_file_preprocessor/txt_processor.c (hunk 3 above)
Comment (Brycey92) [reply spiffs_file_preprocessor/txt_processor.c:63]: oops! you are correct, removing now.

**Review by AEFeinstein — APPROVED** [review]

**Pull request merged**